packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,359 @@
+-   Undefined element types caught at compile time.
+
+Cons:
+
+-   Verbose. Requires user to create `template` element and wrap text in `span`s for styling purposes.
+-   Requires difficult validation to ensure only supported elements are present in the `template`.
+-   Could allow users to provide inline styling.
+-   Blazor: cannot put Blazor components inside `template`--must use raw Nimble elements without type safety
+
+### API
+
+#### Icon column element:
+
+_Component Name_
+
+-   `nimble-table-column-icon`
+
+_Props/Attrs_
+
+-   `field-name`: string
+-   `key-type`: 'string' | 'number' | 'boolean'
+-   `pixel-width`: number (set to the desired fixed column width, else will use a default fixed width)
+
+_Content_
+
+-   column title (text or icon)
+-   1 or more `nimble-mapping-icon` or `nimble-mapping-spinner` elements
+
+#### General mapping column element:
+
+_Component Name_
+
+-   `nimble-table-column-mapping`
+
+_Props/Attrs_
+
+-   `field-name`: string
+-   `key-type`: 'string' | 'number' | 'boolean'
+-   `fractional-width`: number (defaults to 1)
+-   `min-pixel-width`: number (defaults to minimum supported by table)
+
+_Content_
+
+-   column title (text or icon)
+-   1 or more `nimble-mapping-text` elements
+
+#### Mapping element (icon):
+
+_Component Name_
+
+-   `nimble-mapping-icon`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `icon`: string - name of the Nimble icon element
+-   `severity`: string - one of the supported enum values. Controls color of the icon.
+-   `label`: string - localized value used as the accessible name and `title` of the icon. Will also be displayed in the group header.
+-   `default-mapping`: boolean - presence causes this mapping to be used when no others match the value
+
+#### Mapping element (spinner):
+
+_Component Name_
+
+-   `nimble-mapping-spinner`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `label`: string - localized value used as the accessible name and `title` of the spinner. Will also be displayed in the group header.
+-   `default-mapping`: boolean - presence causes this mapping to be used when no others match the value
+
+#### Mapping element (text):
+
+_Component Name_
+
+-   `nimble-mapping-text`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `label`: string - display text
+-   `default-mapping`: boolean - presence causes this mapping to be used when no others match the value
+
+### Anatomy
+
+#### `nimble-table-column-mapping`
+
+```HTML
+<template slot="${x => x.columnInternals.uniqueId}">
+    <slot
+        ${slotted('mappings')}
+        name="mapping">
+    </slot>
+    <span class="header-content">
+        <slot></slot>
+    </span>
+</template>
+```
+
+Cell view:
+
+The cell view relies on the matched mapping to provide a template to render.
+
+```HTML
+<div ${ref('wrapper') class="wrapper"}>
```

Comment:
This might be more of an implementation-PR discussion, but I'm curious why all of this logic is being written in the template. I think it might be more straightforward for the cell view to have a getter that returns the matching mapping that should get rendered in the template. I could be wrong, but that seems like it would be more efficient that having the template have a bunch of conditional logic that's dependent on the state of other parts of the template.

Reply:
That's a fair point. Somehow, I have a bias for doing things in the template, but I don't think I have a justification.